src/EditorFeatures/CSharpTest/CodeActions/IntroduceVariable/IntroduceVariableTests.cs

```diff
@@ -2345,5 +2345,82 @@ void M()
 }
 ");
         }
+
+        [WorkItem(1130990)]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsIntroduceVariable)]
+        public void InParentConditionalAccessExpressions()
+        {
+            var code =
+    @"using System;
+class C
+{
+    public T F<T>(T x)
+    {
+        var y = [|F(new C())|]?.F(new C())?.F(new C());
+        return x;
+    }
+}";
+
+            var expected =
+    @"using System;
+class C
+{
+    public T F<T>(T x)
+    {
+        var {|Rename:c|} = F(new C());
+        var y = c?.F(new C())?.F(new C());
+        return x;
+    }
+}";
+
+            Test(code, expected, index: 0, compareTokens: false);
+        }
+
+        [WorkItem(1130990)]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsIntroduceVariable)]
+        public void MissingAcrossMultipleParentConditionalAccessExpressions()
+        {
+            TestMissing(
+    @"using System;
+class C
+{
+    public T F<T>(T x)
+    {
+        var y = [|F(new C())?.F(new C())|]?.F(new C());
```

Comment:
It's definitely tricky when the selected expression splits one of the non-ConditionalAccess expressions apart. ```[|a?.b|].c``` would be complicated because the right-hand side isn't on a ConditionalAccess boundary, and if you extract ```a?.b``` into something named ```e```, then if ```e``` is null it needs to know whether it was null because ```a``` was null (should cause ```.c``` to be skipped) or if ```b``` was null (should cause ```.c``` to throw a NullReferenceException). 

But, when the selection ends on a ConditionalAccess boundary, then I think we could still do something reasonable. With the selection on a ConditionalAccess boundary, we can simplify to ```e1?.e2?.e3?...``` with no regular member accesses in between. If the specific example of ```[|e1?.e2?.e3|]?.e4```, imagine it already has parentheses around the part we're trying to extract, ```(e1?.e2?.e3)?.e4```. This changes the tree from ```CA<e1, CA<e2, CA<e3, e4>>>``` (where e1, e2, and e3 are split between multiple expressions making it hard to extract) to ```CA<CA<e1, CA<e2, e3>>, e4``` without changing which expressions get evaluated or the order in which they're evaluated<sup>[[1]](#footnote1)</sup> It's then easy to extract the part we want.

If we did this, we'd have to add some ConditionalAccess-specific handling to the Find All Occurrences fixer, as it would again involve traversing the ConditionalAccess tree structure.

<a name="footnote1">[1]</a> When e2 is null, it would previously skip the entire rest of the expression all at once, but now it would have to skip it in two phases, once to complete the parenthesized expression, and once again to skip the rest of the larger conditional access, but this probably only slows things down on very hot code paths.